File: auth/internal/utils/sendMail.go
package utils

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"

	model "github.com/p-society/gc-server/schemas/pkg/models"
)

func SendEmail(p *model.Player) error {
	var (
		url         = "http://localhost:6969/v0/mails"
		requestBody = map[string]interface{}{
			"subject": "Grand Championship Player Verification",
			"content": RenderEngine(p.OTP),
			"to":      []string{p.Email},
		}
	)

	requestBodyBytes, err := json.Marshal(requestBody)
	if err != nil {
		return fmt.Errorf("error encoding JSON: %v", err)
	}

	resp, err := http.Post(url, "application/json", bytes.NewBuffer(requestBodyBytes))
	if err != nil {
		return fmt.Errorf("error sending request: %v", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("unexpected response status: %s", resp.Status)
	}

	return nil
}

File: apis/src/utils/sendMail.ts
export default async function sendMail(email: string, OTP: number) {
  const mailUri = 'http://127.0.0.1:6969/v0/mails';
  const postData = {
    'subject': 'Grand Championship Player Verification',
    'content': OTP.toString(),
    'to': [email],
  };

  const requestOptions = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(postData) // Convert the data object to a JSON string
  };

  await fetch(mailUri, requestOptions)
    .then(response => {
      if (!response.ok) {
        throw new Error('Network response was not ok');
      }
      return response.json(); // Parse the JSON response
    });
}